plugin/cr/src/test/java/org/opencds/cqf/ruler/cr/r4/RepositoryServiceTest.java

```diff
@@ -1153,5 +1156,86 @@ void packageOperation_big_bundle() {
+		Parameters supplementalBundleParams = parameters(
+			part("resource", ersdExampleSupplementalBundle)
+		);
+		OperationOutcome supplementalBundleOutcome = getClient().operation()
+			.onServer()
+			.named("$validate")
+			.withParameters(supplementalBundleParams)
+			.returnResourceType(OperationOutcome.class)
+			.execute();
+		List<OperationOutcomeIssueComponent> supplementalBundleErrors = supplementalBundleOutcome.getIssue().stream().filter((issue) -> issue.getSeverity() == IssueSeverity.ERROR || issue.getSeverity() == IssueSeverity.FATAL).collect(Collectors.toList());
+		assertTrue(supplementalBundleErrors.size() == 0);
+
+		Library validationErrorLibrary = (Library) loadResource("ersd-active-library-us-ph-validation-failure-example.json");
+		Parameters validationFailedParams = parameters(
+			part("resource", validationErrorLibrary)
+		);
+		OperationOutcome failedValidationOutcome = getClient().operation()
+			.onServer()
+			.named("$validate")
+			.withParameters(validationFailedParams)
+			.returnResourceType(OperationOutcome.class)
+			.execute();
+		List<OperationOutcomeIssueComponent> invalidLibraryErrors = failedValidationOutcome.getIssue().stream().filter((issue) -> issue.getSeverity() == IssueSeverity.ERROR || issue.getSeverity() == IssueSeverity.FATAL).collect(Collectors.toList());
+		assertTrue(invalidLibraryErrors.size() == 5);
+
+		Parameters noResourceParams = parameters();
+		UnprocessableEntityException noResourceException = null;
+		try {	
+			getClient().operation()
+				.onServer()
+				.named("$validate")
+				.withParameters(noResourceParams)
+				.returnResourceType(OperationOutcome.class)
+				.execute();
+		} catch (UnprocessableEntityException e) {
+			noResourceException = e;
+		}
+		assertNotNull(noResourceException);
+		assertTrue(noResourceException.getMessage().contains("resource must be provided"));
+	}
+	@Test
+	void validatePackageOutput() {
+		loadTransaction("ersd-active-transaction-bundle-example.json");
+		Bundle packagedBundle = getClient().operation()
+			.onInstance(specificationLibReference)
+			.named("$crmi.package")
+			.withParameters(parameters())
+			.returnResourceType(Bundle.class)
+			.execute();
+		Parameters packagedBundleParams = parameters(
+			part("resource", packagedBundle)
+		);
+		OperationOutcome packagedBundleOutcome = getClient().operation()
+			.onServer()
+			.named("$validate")
+			.withParameters(packagedBundleParams)
+			.returnResourceType(OperationOutcome.class)
+			.execute();
+		List<OperationOutcomeIssueComponent> packagedBundleValidationErrors = packagedBundleOutcome.getIssue().stream().filter((issue) -> issue.getSeverity() == IssueSeverity.ERROR || issue.getSeverity() == IssueSeverity.FATAL).collect(Collectors.toList());
+		// currently the packaged bundle is not valid due to:
+		// - bundle.total
+		// - bundle.entry.request
+		// being present on a bundle with type = collection
```

Comment:
may need new profiles to avoid these validation failures

Reply:
Can't profile at the bundle level because of the possibility that the package will span multiple bundles.

Reply:
👍🏾 I'll split the test into individual validations on example resources to match our future workflow

Reply:
back to validating whole bundle based on discussion